backend/internal/controller/adapter/dispatcher.go

```diff
@@ -0,0 +1,68 @@
+package adapter
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type Category int
+
+const (
+	None Category = iota
+	Initial
+	Goal
+	Position
+)
+
+type Message struct {
+	Category Category
+	TableID  string
+	Team     int
+}
+
+type dispatcherMsg struct {
+	MsgType   string  `json:"type,omitempty"`
+	Origin    string  `json:"origin,omitempty"`
+	TableID   string  `json:"id,omitempty"`
+	X         float64 `json:"x,omitempty"`
+	Y         float64 `json:"y,omitempty"`
+	Timestamp string  `json:"timestamp,omitempty"`
+	Goal      int     `json:"goal,omitempty"`
+	Sequence  string  `json:"Sequence,omitempty"`
+}
+
+type initialResponse struct {
+	GameID    string `json:"start,omitempty"`
+	GameEnded int    `json:"end,omitempty"`
+}
+
+func Unpack(message io.Reader) (*Message, error) {
+	var tableMessage dispatcherMsg
+
+	err := json.NewDecoder(message).Decode(&tableMessage)
+	if err != nil {
+		return &Message{}, err
```

Comment:
```suggestion
		return nil, err
```

Reply:
We are not able to return nil in this function. Instead we will return an empty message and error.